### PR TITLE
CreateUnit improvements

### PR DESCRIPTION
**Pull request by Starkku**
- `CreateUnit.ConsiderPathfinding`, if set, makes game look for eligible cells around the animation's coordinates if the cell at those coordinates is not passable to the created unit, and creates the unit at the first eligible cell found instead.
- Warhead `AnimList/SplashList` animations and animations created through map trigger action `41 Play Anim At` now have owner set correctly if the animation has `CreateUnit` set.

### CODE REVIEW

**Comment by CommanderNour** [comment]
Tested & working correctly

**Review by YRIDZE — CHANGES_REQUESTED** [review]
Please, check my comments and fix the issues. Add info to "What`s New".

**Comment by Starkku** [comment]
> Please, check my comments and fix the issues. Add info to "What`s New".

This particular pull request does not warrant a separate entry in "What's New" as the only changes contained are additions to a feature that is being introduced in the version the "What's New" is currently comprised of.

File: src/Ext/AnimType/Hooks.AnimCreateUnit.cpp [hunk 1]
```diff
@@ -126,4 +144,54 @@ DEFINE_HOOK(0x424932, AnimClass_Update_CreateUnit_ActualAffects, 0x6)
...
+	GET_STACK(HouseClass*, pHouse, STACK_OFFS(0x18, -0x4));
+
+	if (pAnim)
+	{
+		auto const pTypeExt = AnimTypeExt::ExtMap.Find(pAnim->Type);
+
+		if (auto unit = pTypeExt->CreateUnit.Get())
+		{
+			AnimExt::SetAnimOwnerHouseKind(pAnim, pHouse, pHouse, pHouse);
+		}
```
Comment (Metadorius) [inline src/Ext/AnimType/Hooks.AnimCreateUnit.cpp:193]:
```suggestion
		if (auto unit = pTypeExt->CreateUnit.Get())
			AnimExt::SetAnimOwnerHouseKind(pAnim, pHouse, pHouse, pHouse);

```

File: src/Ext/AnimType/Hooks.AnimCreateUnit.cpp [hunk 2]
```diff
@@ -126,4 +144,54 @@ DEFINE_HOOK(0x424932, AnimClass_Update_CreateUnit_ActualAffects, 0x6)
...
+	{
+		auto const pTypeExt = AnimTypeExt::ExtMap.Find(pAnim->Type);
+
+		HouseClass* pInvoker = (pThis->Owner) ? pThis->Owner->Owner : nullptr;
+		HouseClass* pVictim = nullptr;
+
+		if (TechnoClass* Target = generic_cast<TechnoClass*>(pThis->Target))
+		{
+			pVictim = Target->Owner;
+		}
```
Comment (Metadorius) [inline src/Ext/AnimType/Hooks.AnimCreateUnit.cpp:166]:
```suggestion
		if (TechnoClass* Target = generic_cast<TechnoClass*>(pThis->Target))
			pVictim = Target->Owner;
```

File: src/Ext/AnimType/Hooks.AnimCreateUnit.cpp [hunk 3]
```diff
@@ -126,4 +144,54 @@ DEFINE_HOOK(0x424932, AnimClass_Update_CreateUnit_ActualAffects, 0x6)
...
+
+		if (auto unit = pTypeExt->CreateUnit.Get())
+		{
+			AnimExt::SetAnimOwnerHouseKind(pAnim, pInvoker, pVictim, pInvoker);
+		}
```
Comment (Metadorius) [inline src/Ext/AnimType/Hooks.AnimCreateUnit.cpp:171]:
```suggestion
		if (auto unit = pTypeExt->CreateUnit.Get())
			AnimExt::SetAnimOwnerHouseKind(pAnim, pInvoker, pVictim, pInvoker);
```

**Review by Metadorius — APPROVED** [review]

**Pull request merged**